### PR TITLE
Model2 Approach to a Unified Common-Utils

### PR DESCRIPTION
**Pull request by okram**
This is related to ISSUE-10 of security analytics. It's the best push I have that uses a simpler Model infrastructure based on public fields and reflection. It also incorporates the notion of redirecting existing actions to a intermediate "translator action" (@getsaurabh02 idea). 

This is a 3 part PR across common-utils, alerting, and security analytics. All under the same `model2/` branch.

*CheckList:*
[X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/commons/model2/ModelSerializer.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ModelSerializer {
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/commons/model2/ModelSerializer.java:25]: Dont see a need for ModelSerializer now.

File: src/main/java/org/opensearch/commons/model2/action/ExecuteMonitorAction.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.opensearch.commons.model2.action;
+
+import org.opensearch.action.ActionType;
+
+public class ExecuteMonitorAction extends ActionType<ExecuteMonitorResponse> {
+
+    public static final String ALERTING_NAME = "cluster:admin/opendistro/security_analytics/monitor/execute";
+    public static final ExecuteMonitorAction ALERTING_INSTANCE = new ExecuteMonitorAction(ALERTING_NAME);
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/commons/model2/action/ExecuteMonitorAction.java:8]: Can we only move the ACTION_NAME here in the common-utils while the Action class can still be retailed by Alerting. That way ownership remains clear, while the action name is available to clients.

File: src/main/java/org/opensearch/commons/model2/action/ExecuteMonitorRequest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.commons.model2.ModelSerializer;
+import org.opensearch.commons.model2.model.Monitor;
+
+import java.io.IOException;
+
+public class ExecuteMonitorRequest extends ActionRequest {
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/commons/model2/action/ExecuteMonitorRequest.java:13]: Can we create a dedicated sub-project (common path) for all alerting related models

File: src/main/java/org/opensearch/commons/model2/action/ExecuteMonitorResponse.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    @Override
+    public void writeTo(final StreamOutput output) throws IOException {
+        ModelSerializer.write(output, this.monitor);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder xContentBuilder, ToXContent.Params params) throws IOException {
+        return ModelSerializer.write(xContentBuilder, this.monitor);
+        // return xContentBuilder.startObject().field(Tokens.MONITOR_ID, this.monitorId).endObject();
+    }
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/commons/model2/action/ExecuteMonitorResponse.java:39]: Every models should be hold its our own StreamInput and StreamOutput serialization logic.

File: src/main/java/org/opensearch/commons/model2/model/Action.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    public Throttle throttle;
+    public ExecutionPolicy action_execution_policy;
+
+    //val throttleEnabled: Boolean,
+    // val throttle: Throttle?,
+    //val id: String = UUIDs.base64UUID(),
+    //val actionExecutionPolicy: ActionExecutionPolicy? = null
+    public Action() {
+        // for serialization
+    }
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/commons/model2/model/Action.java:32]: I see a few ToDo's across the model classes. Are we planning to update them as part of this PR incrementally ?

**Comment by lezzago** [comment]
Another approach has been chosen, so closing the PR

**Pull request closed without merging**